rewrite-yaml/src/main/java/org/openrewrite/yaml/YamlParser.java

```diff
@@ -106,9 +106,12 @@ private Yaml.Documents parseFromInput(Path sourceFile, EncodingDetectingInputStr
             yamlSourceWithVariablePlaceholders.append(yamlSource, pos, yamlSource.length());
         }
 
-        try (FormatPreservingReader reader = new FormatPreservingReader(
-                new InputStreamReader(new ByteArrayInputStream(yamlSourceWithVariablePlaceholders.toString()
-                        .getBytes(StandardCharsets.UTF_8))))) {
+        try (
```

Comment:
I suggest we skip the entire encoding-decoding and just use a `StringReader`. Something like:
```java
       try (FormatPreservingReader reader = new FormatPreservingReader(
                new StringReader(yamlSourceWithVariablePlaceholders.toString()))) {
```

Reply:
Good point, I change it.